### ISSUE TITLE
Error compiling under macOS Catalina: no template named 'conditional_t'

### ISSUE THREAD

**Issue by hendrikkayser** [gisogrimm/ov-client#33]
Hello,
I'm trying to compile the ov-client under macOS 10.15.7 following the steps provided by the Github Actions in the "MacOS make" workflow. The make command called in build_osx.sh fails with the follwoing errors. Any hints? 
 
```
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:54:12: error: no template named 'conditional_t' in namespace 'std'; did you mean 'conditional'?
    : std::conditional_t<Trait::value, conjunction<Traits...>, Trait>
      ~~~~~^~~~~~~~~~~~~
           conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:530:33: note: 'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:68:12: error: no template named 'conditional_t' in namespace 'std'; did you mean 'conditional'?
    : std::conditional_t<Trait::value, Trait, disjunction<Traits...>>
      ~~~~~^~~~~~~~~~~~~
           conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:530:33: note: 'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:95:23: error: no template named 'remove_cv_t' in namespace 'std'; did you mean 'remove_cv'?
    using type = std::remove_cv_t<std::remove_reference_t<T>>;
                 ~~~~~^~~~~~~~~~~
                      remove_cv
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:697:50: note: 'remove_cv' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_cv
                                                 ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:95:35: error: no template named 'remove_reference_t' in namespace 'std'; did you mean simply 'remove_reference_t'?
    using type = std::remove_cv_t<std::remove_reference_t<T>>;
                                  ^~~~~~~~~~~~~~~~~~~~~~~
                                  remove_reference_t
/usr/local/include/boost/type_traits/remove_reference.hpp:53:23: note: 'remove_reference_t' declared here
   template <class T> using remove_reference_t = typename remove_reference<T>::type;
                      ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:105:18: error: no template named 'remove_const_t' in namespace 'std'; did you mean simply 'remove_const_t'?
    using type = std::remove_const_t<std::remove_reference_t<T>>;
                 ^~~~~~~~~~~~~~~~~~~
                 remove_const_t
/usr/local/include/boost/type_traits/remove_const.hpp:33:23: note: 'remove_const_t' declared here
   template <class T> using remove_const_t = typename remove_const<T>::type;
                      ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:105:38: error: no template named 'remove_reference_t' in namespace 'std'; did you mean simply 'remove_reference_t'?
    using type = std::remove_const_t<std::remove_reference_t<T>>;
                                     ^~~~~~~~~~~~~~~~~~~~~~~
                                     remove_reference_t
/usr/local/include/boost/type_traits/remove_reference.hpp:53:23: note: 'remove_reference_t' declared here
   template <class T> using remove_reference_t = typename remove_reference<T>::type;
                      ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:115:18: error: no template named 'remove_const_t' in namespace 'std'; did you mean simply 'remove_const_t'?
    using type = std::remove_const_t
                 ^~~~~~~~~~~~~~~~~~~
                 remove_const_t
/usr/local/include/boost/type_traits/remove_const.hpp:33:23: note: 'remove_const_t' declared here
   template <class T> using remove_const_t = typename remove_const<T>::type;
                      ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:117:18: error: no template named 'remove_pointer_t' in namespace 'std'; did you mean 'remove_pointer'?
            std::remove_pointer_t<std::remove_reference_t<T>>
            ~~~~~^~~~~~~~~~~~~~~~
                 remove_pointer
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1189:50: note: 'remove_pointer' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS remove_pointer                      {typedef _LIBCPP_NODEBUG_TYPE _Tp type;};
                                                 ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:117:35: error: no template named 'remove_reference_t' in namespace 'std'; did you mean simply 'remove_reference_t'?
            std::remove_pointer_t<std::remove_reference_t<T>>
                                  ^~~~~~~~~~~~~~~~~~~~~~~
                                  remove_reference_t
/usr/local/include/boost/type_traits/remove_reference.hpp:53:23: note: 'remove_reference_t' declared here
   template <class T> using remove_reference_t = typename remove_reference<T>::type;
                      ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:129:23: error: no template named 'conditional_t' in namespace 'std'; did you mean 'conditional'?
    using type = std::conditional_t
                 ~~~~~^~~~~~~~~~~~~
                      conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:530:33: note: 'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:131:39: error: no template named 'remove_reference_t' in namespace 'std'; did you mean simply 'remove_reference_t'?
                        std::is_const<std::remove_reference_t<From>>::value,
                                      ^~~~~~~~~~~~~~~~~~~~~~~
                                      remove_reference_t
/usr/local/include/boost/type_traits/remove_reference.hpp:53:23: note: 'remove_reference_t' declared here
   template <class T> using remove_reference_t = typename remove_reference<T>::type;
                      ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:132:30: error: no template named 'add_const_t' in namespace 'std'; did you mean 'add_const'?
                        std::add_const_t<To>,
                        ~~~~~^~~~~~~~~~~
                             add_const
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1080:50: note: 'add_const' declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS add_const {
                                                 ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
In file included from /usr/local/include/boost/geometry/core/ring_type.hpp:28:
In file included from /usr/local/include/boost/geometry/core/tag.hpp:23:
/usr/local/include/boost/geometry/util/type_traits_std.hpp:166:18: error: no template named 'conditional_t' in namespace 'std'; did you mean 'conditional'?
    typedef std::conditional_t
            ~~~~~^~~~~~~~~~~~~
                 conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:530:33: note: 'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
/usr/local/include/boost/geometry/core/ring_type.hpp:96:18: error: no template named 'conditional_t' in namespace 'std'; did you mean 'conditional'?
    typedef std::conditional_t
            ~~~~~^~~~~~~~~~~~~
                 conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:530:33: note: 'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
/usr/local/include/boost/geometry/core/ring_type.hpp:111:18: error: no template named 'conditional_t' in namespace 'std'; did you mean 'conditional'?
            std::conditional_t
            ~~~~~^~~~~~~~~~~~~
                 conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:530:33: note: 'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:34:
In file included from /usr/local/include/boost/geometry/core/closure.hpp:24:
/usr/local/include/boost/geometry/core/ring_type.hpp:127:18: error: no template named 'conditional_t' in namespace 'std'; did you mean 'conditional'?
            std::conditional_t
            ~~~~~^~~~~~~~~~~~~
                 conditional
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:530:33: note: 'conditional' declared here
    struct _LIBCPP_TEMPLATE_VIS conditional {typedef _If type;};
                                ^
In file included from src/receivermod_itu51.cc:1:
In file included from /usr/local/include/boost/geometry.hpp:17:
In file included from /usr/local/include/boost/geometry/geometry.hpp:35:
/usr/local/include/boost/geometry/core/coordinate_dimension.hpp:101:23: error: constexpr function's return type 'void' is not a literal type
constexpr inline void assert_dimension()
                      ^
/usr/local/include/boost/geometry/core/coordinate_dimension.hpp:111:23: error: constexpr function's return type 'void' is not a literal type
constexpr inline void assert_dimension_less_equal()
                      ^
/usr/local/include/boost/geometry/core/coordinate_dimension.hpp:117:23: error: constexpr function's return type 'void' is not a literal type
constexpr inline void assert_dimension_greater_equal()
                      ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
2 warnings and 20 errors generated.
make[1]: *** [build/ovclienttascarreceiver_itu51.dylib] Error 1
make: *** [tscplug] Error 2
```

**Comment by gisogrimm** [comment]
This is a problem with boost geometry versions. The itu5.1 receiver type can be deactivated on MacOS for the time being.

**Comment by gisogrimm** [comment]
solved with commit dd69cfc95ef
